src/OpenSage.Game/Game.cs

```diff
@@ -853,6 +853,30 @@ private void LocalLogicTick(IEnumerable<InputMessage> messages)
             // We pass RenderTime to Scene2D so that the UI remains responsive even when the game is paused.
             Scene2D.LocalLogicTick(RenderTime, Scene3D?.LocalPlayer);
             Scene3D?.LocalLogicTick(MapTime, tickT);
+
+            // TODO: do this properly (this is a hack to call StartMultiplayerGame on the correct thread)
+            if (SkirmishManager.SkirmishGame?.ReadyToStart == true)
+            {
+                SkirmishManager.SkirmishGame.ReadyToStart = false;
+                StartMultiPlayerGame(
```

Comment:
This code is very dense (pattern matching inside a constructor inside a LINQ expression inside an argument list) - I would personally split it into few more lines, at least by moving the player settings to a separate variable.

Reply:
I second what @paavohuhtala said.

Reply:
@tgjones we've already split playerSettings into a seperate variable, but this probably still should be simpler